### DISCUSSION TITLE
Switching from GPG to Minisign for download signatures

### DISCUSSION THREAD

**Discussion by samuel-lucas6** [samuel-lucas6/Kryptor#12] [Announcements]
I'd like to stop using GPG for the download signatures because it's an unfriendly and dated program. However, I can foresee people complaining about GPG signatures not being used since GPG is preinstalled on many Linux distros. With that said, GPG is less accessible for macOS and Windows users.

The obvious alternative is [Minisign](https://github.com/jedisct1/minisign), but there aren't pre-built binaries for Linux. I believe certain distros offer downloads though. I could also use Kryptor signatures (coming in v3-beta) like how Frank Denis signs his Minisign binaries using Minisign, but then a user can't verify the download without already having Kryptor, which doesn't make much sense.

If you have any thoughts, please share them. People have got to stop using GPG at some point and move on to better tools. The one thing I don't want to do is sign things with multiple tools because that's extra, unnecessary work putting out each release.

### Update (03/01/22)
For the record, my Minisign public key is `RWRZzvSG5htEUFlSEl88cMb0pTGXSZmzqTujwpAweUrSJQuWYui8Qpk9`. However, I've decided to switch to Kryptor signatures for future releases to allow automated `-u|--update` functionality and to save people having to download another program.

**Comment by covert-encryption** [comment]
One pretty neat idea is to download keys from Github. This is implemented in Age and in our tool Covert where you can simply specify a recipient with `-R github:samuel-lucas6` to have the program fetch any SSH keys you may have uploaded here. You can convert MiniSign keys into SSH keys and vice-versa. They are the same, just encoded slightly different.

Another possibility for key distribution is to attach the keys to each message sent. This should be quite natural, avoiding the separate step of getting keys from some place else, at least after the initial contact is made. Given that modern keys (Ed25519 in Minisign format) are very short, they are also easily attached to profile pages, Git commits (instead of email) or even business cards... And the same keys can be used for both encryption and signatures, no reason to have dual keys.
